qcloud_sms/models/qcloud_sms.py

```diff
@@ -0,0 +1,255 @@
+        sms = self.create(vals)
+
+        # get SMS object
+        qcloudsms = self.env['ir.config_parameter'].get_qcloud_sms_object()
+        msender = qcloudsms.SmsMultiSender()
+
+        try:
+            phone_obj_list = map(self._sms_sanitization, partners)
+        except UserError as e:
+            sms.state = 'error'
+            _logger.debug(e)
+
+        country_code_list = list(map(lambda x: x.country_code, phone_obj_list))
+        country_code = list(set(country_code_list))
+
+        if len(country_code) > 1:
+            raise UserError(_('The country code must be the same for all phone numbers'))
+
+        country_code = country_code[0]
+        national_number_list = list(map(lambda x: x.national_number, phone_obj_list))
+
+        _logger.debug("Country code: %s, Mobile numbers: %s", country_code, national_number_list)
+
+        sms_type = sms.template_id.sms_type if sms.template_id else 0
+
+        # China Country Code 86 (use domestics templates)
+        if country_code == '86':
+            template_ID = sms.template_id.domestic_sms_template_ID if sms.template_id else False
+            params = sms.template_id.domestic_template_params if sms.template_id else False
+            sign = sms.template_id.domestic_sms_sign if sms.template_id else False
+        else:
+            template_ID = sms.template_id.international_sms_template_ID if sms.template_id else False
+            params = sms.template_id.international_template_params if sms.template_id else False
+            sign = sms.template_id.international_sms_sign if sms.template_id else False
+
+        if params:
+            params = params.split(',')
+
+        extend_field = kwargs.get('extend_field') or ""
+
+        if template_ID or sign:
+            # send sms by params
+            result = msender.send_with_param(country_code, national_number_list, template_ID, params, sign=sign,
+                                             extend=extend_field, ext=sms.id)
+        else:
+            result = msender.send(sms_type, country_code, national_number_list,
+                                  message, extend=extend_field, ext=sms.id)
+
+        if result.get('result') == 0:
+            sms.state = 'sent'
+        else:
+            sms.state = 'error'
+
+        return result
+
+
+class QCloudSMSTemplate(models.Model):
+    """Templates of SMS messages."""
+
+    _name = 'qcloud.sms.template'
+    _description = 'SMS Templates'
+    _order = 'id desc'
+
+    name = fields.Char(string='Name', help='The template name')
+
+    domestic_sms_template_ID = fields.Integer(
+        string='Domestic SMS Template ID',
+        help='SMS Template ID is the Tencent Cloud SMS template (the specific content of the SMS message to be sent).'
+    )
+    domestic_template_params = fields.Text(
+        string="Domestic Template parameters",
+        help="Parameters must be separated by commas. If the template has no parameters, leave it empty."
+    )
```

Comment:
It's not clear how to use params and what is params

Reply:
@yelizariev I have only this description of params https://cloud.tencent.com/document/product/382/5976

Reply:
@yelizariev params - array of template values. For example:
params = ['12345', '5']
template = "Your login verification code is {1}, which is valid for {2} minutes."

result = "Your login verification code is 12345, which is valid for 5 minutes."

Reply:
ok, but this explanation should be somewhere in fields description or at the form